app-android/src/main/java/com/ustadmobile/port/android/view/ChatListFragment.kt

```diff
@@ -0,0 +1,157 @@
+        super.onViewCreated(view, savedInstanceState)
+        fabManager?.text = requireContext().getText(R.string.chat)
+        //override this to show our own bottom sheet
+        fabManager?.onClickListener = {
+            val optionList =
+                listOf(
+                    BottomSheetOption(R.drawable.ic_add_black_24dp,
+                        requireContext().getString(R.string.new_chat), NEW_CHAT),
+                    BottomSheetOption(R.drawable.ic_login_24px,
+                    requireContext().getString(R.string.new_group), NEW_GROUP)
+                )
+
+            val sheet = OptionsBottomSheetFragment(optionList, this)
+            sheet.show(childFragmentManager, sheet.tag)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        menu.findItem(R.id.menu_search).isVisible = true
+    }
+
+    override fun onResume() {
+        super.onResume()
+        mActivityWithFab?.activityFloatingActionButton?.text =
+                requireContext().getString(R.string.chat)
+    }
+
+    /**
+     * OnClick function that will handle when the user clicks to create a new item
+     */
+    override fun onClick(v: View?) {
+        if(v?.id == R.id.item_createnew_layout) {
+           mPresenter?.handleClickAddNewItem()
+        }else{
+            super.onClick(v)
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        mPresenter = null
+        dbRepo = null
+        mDataRecyclerViewAdapter = null
+        mDataBinding = null
+    }
+
+    override val displayTypeRepo: Any?
+        get() = dbRepo?.chatDao
+
+    override fun onBottomSheetOptionSelected(optionSelected: BottomSheetOption) {
+        when(optionSelected.optionCode) {
+            NEW_CHAT -> mPresenter?.handleClickCreateNewFab(
+                ChatListPresenter.CHAT_RESULT_KEY)
+            NEW_GROUP -> mPresenter?.handleClickCreateNewGroup(
+                arguments.toStringMap(),
+                ChatListPresenter.CHAT_RESULT_KEY_GROUP)
+        }
+    }
+
+    companion object {
+        val DIFF_CALLBACK: DiffUtil.ItemCallback<ChatWithLatestMessageAndCount> = object
+            : DiffUtil.ItemCallback<ChatWithLatestMessageAndCount>() {
+            override fun areItemsTheSame(oldItem: ChatWithLatestMessageAndCount,
+                                         newItem: ChatWithLatestMessageAndCount): Boolean {
+
+                return oldItem.chatUid == newItem.chatUid
+            }
+
+            override fun areContentsTheSame(oldItem: ChatWithLatestMessageAndCount,
+                                            newItem: ChatWithLatestMessageAndCount): Boolean {
+                return oldItem == newItem
```

Comment:
The diffutil for a read-only item should only compare fields that are displayed to the user. That is important to prevent all the existing chat message viewholders being redrawn needlessly when a new chat message lands.